src/main/java/com/google/step/datamanager/RestaurantManagerDatastore.java

```diff
@@ -81,4 +83,17 @@ private Restaurant transformEntitytoRestaurant(Entity restaurantEntity) {
     long id = restaurantEntity.getKey().getId();
     return new Restaurant(id, name, photoBlobkey);
   }
+
+  @Override
+  public List<Restaurant> readRestaurants(List<Long> ids) {
+    List<Restaurant> restaurants = new ArrayList<>();
+    for (long id : ids) {
+      try {
+        restaurants.add(readRestaurant(id));
```

Comment:
Doing all the lookups this way sequentially will increase the latency of the api as well as the load on the database. Just lookup all ids at once with `datastore.get(ids)`. Batch calls are usually faster and more efficient than several single lookups.
API reference - https://cloud.google.com/appengine/docs/standard/java/javadoc/com/google/appengine/api/datastore/DatastoreService.html

Reply:
okay I see.